### PR TITLE
doc: document our versioning process

### PR DESCRIPTION
**Pull request by monperrus**
Our versioning convention is not documented. Here it is.

### CODE REVIEW

File: doc/_release/README.md [hunk 1]
```diff
@@ -1,4 +1,14 @@
-# How to release Spoon?
+# Spoon Releases
+
+## Versioning
+
+Spoon uses a three digit version number MAJOR.MINOR.HOTPATCH
+
+* we bump MAJOR when there is a big new feature or a strongly breaking change. We use it mostly for new features since we take special care to backward compatibility.
+* we bump HOTPATCH when a normal release is not operational.
```
Comment (MartinWitt) [inline doc/_release/README.md:8]: Maybe change the order of the points to follow the order above `MAJOR.MINOR.HOTPATCH` ?

**Comment by slarse** [comment]
The only difference between this versioning and SemVer 2 is that we also increment the major version number for significant new features. IMO we could replace this text with: "We follow semantic versioning, with the exception that we may increment the major version number for significant new features even when there are no breaking changes"

**Comment by monperrus** [comment]
added @slarse 's proposed sentence and switched the two bullets per @MartinWitt 's suggestion.

ready to merge.

**Comment by MartinWitt** [comment]
Thanks @monperrus

**Pull request merged**